### PR TITLE
fix: essence tranche token metadata

### PR DESCRIPTION
**Pull request by wischli**
# Description

[Slack thread](https://kflabs.slack.com/archives/C04GJQAM9P0/p1712041396660229)

We can hold this until #1756 is merged

## Changes and Descriptions

* Fixes incorrect tranche token metadata reported by pool system `essence` which is used for `Updated` event
* Adds missing unit test which asserts condition

# Checklist:

- [x] I have added Rust doc comments to structs, enums, traits and functions
- [x] I have made corresponding changes to the documentation
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: pallets/pool-system/src/pool_types.rs [hunk 1]
```diff
@@ -300,16 +303,12 @@ impl<
...
-					location: None,
-					additional: (),
-				})
-				.unwrap();
+			let metadata = AssetRegistry::metadata(
+				&<AssetRegistry as orml_traits::asset_registry::Inspect>::AssetId::from(
+					tranche.currency.into(),
+				),
+			)
+			.ok_or(DispatchError::CannotLookup)?;
```
Comment (lemunozm) [inline pallets/pool-system/src/pool_types.rs:311]: Thanks for cleaning that weird `ok_or` and removing the `unwrap()`

File: pallets/pool-system/src/tests/mod.rs [hunk 2]
```diff
@@ -2439,6 +2439,69 @@ fn create_tranche_token_metadata() {
...
+		let pool_owner = 1_u64;
+		let tranche_input = vec![
+			TrancheInput {
+				tranche_type: TrancheType::Residual,
+				seniority: None,
+				metadata: TrancheMetadata {
+					token_name: BoundedVec::try_from("ResName".as_bytes().to_owned())
+						.expect("String not out of bounds"),
+					token_symbol: BoundedVec::try_from("ResSym".as_bytes().to_owned())
+						.expect("String not out of bounds"),
```
Comment (lemunozm) [inline pallets/pool-system/src/tests/mod.rs:2454]: NIT. If the test does not check `token_name` or `token_symbol` you can avoid initializing them using directly `BoundedVec::default()` to ease the reading of the test

**Review by lemunozm — APPROVED** [review]
Thanks for taking care of this fix @wischli and add tests related to it 💯 

Just one NIT but LGTM!

File: pallets/pool-system/src/tests/mod.rs (hunk 2 above)
Comment (wischli) [reply pallets/pool-system/src/tests/mod.rs:2454]: We are checking the tranche metadata in L2498 which is why I did not want to use default values here

File: pallets/pool-system/src/pool_types.rs (hunk 1 above)
Comment (wischli) [reply pallets/pool-system/src/pool_types.rs:311]: Yeah that code path gave me shivers

File: pallets/pool-system/src/tests/mod.rs (hunk 2 above)
Comment (lemunozm) [reply pallets/pool-system/src/tests/mod.rs:2454]: Ah ok, I didn't see it. Then everything is perfect!

**Review by mustermeiszer — APPROVED** [review]

**Review by lemunozm — APPROVED** [review]
Reapproving!

**Pull request merged**